controllers/keda/kedacontroller_controller.go

```diff
@@ -379,6 +397,28 @@ func (r *KedaControllerReconciler) installController(logger logr.Logger, instanc
 	return nil
 }
 
+// installMonitoring install the controller resources for the monitoring stack
+func (r *KedaControllerReconciler) installMonitoring(ctx context.Context, logger logr.Logger, instance *kedav1alpha1.KedaController) error {
+	logger.Info("Reconciling monitoring resources")
+
+	// this works only if required CRDs are present
+	if !util.HasServiceMonitorCRD(ctx, logger, r.Client) {
+		logger.Info("ServiceMonitor CRD not found, skipping monitoring resources")
```

Comment:
How often does this function get called from the apply loop? I'm worried about log spam in clusters which don't have the CRDs.

Reply:
Good point; this log message will appear on every failed reconciliation loop. Having it last in the main execution flow narrows the occurrences to failures in [util.UpdateKedaControllerStatus](https://github.com/kedacore/keda-olm-operator/blob/9dab6d3c1943d0555c926b429ea80147db0f7bb9/controllers/keda/kedacontroller_controller.go#L211) which is a `Patch` API call. The happy path, the log will appears only once during the first reconciliation.

Reply:
I was just worried about the Patch call being made on each reconciliation loop iteration since the objects still wouldn't exist. As discussed on slack, we're considering raising the V(erbosity) level so it doesn't log by default.

Reply:
Adjusted.